eclass/libretro-core.eclass

```diff
@@ -0,0 +1,162 @@
+# @DESCRIPTION:
+# Name of this Libretro core. The libretro-core_src_install() phase function
+# will install the shared library "${S}/${LIBRETRO_CORE_NAME}_libretro.so" as a
+# Libretro core. Defaults to the name of the current package excluding the
+# "libretro-" prefix (e.g., "mgba" for the package "libretro-mgba").
+: ${LIBRETRO_CORE_NAME:=${PN#libretro-}}
+
+# @ECLASS-VARIABLE: LIBRETRO_COMMIT_SHA
+# @DESCRIPTION:
+# Commit SHA used for SRC_URI will die if not set in <9999 ebuilds.
+# Needs to be set before inherit.
+
+# @ECLASS-VARIABLE: LIBRETRO_REPO_NAME
+# @REQUIRED
+# @DESCRIPTION:
+# Contains the real repo name of the core formatted as "repouser/reponame".
+# Needs to be set before inherit. Otherwise defaults to "libretro/${PN}"
+: ${LIBRETRO_REPO_NAME:="libretro/libretro-${LIBRETRO_CORE_NAME}"}
+
+: ${HOMEPAGE:="https://github.com/${LIBRETRO_REPO_NAME}"}
+
+if [[ ${PV} == "9999" ]]; then
+	: ${EGIT_REPO_URI:="https://github.com/${LIBRETRO_REPO_NAME}.git"}
+	inherit git-r3
+else
+	[ -z "${LIBRETRO_COMMIT_SHA}" ] && die "LIBRETRO_COMMIT_SHA must be set before inherit."
+	S="${WORKDIR}/${LIBRETRO_REPO_NAME##*/}-${LIBRETRO_COMMIT_SHA}"
+	: ${SRC_URI:="https://github.com/${LIBRETRO_REPO_NAME}/archive/${LIBRETRO_COMMIT_SHA}.tar.gz -> ${P}.tar.gz"}
+fi
+inherit flag-o-matic
+
+# @ECLASS-VARIABLE: LIBRETRO_CORE_LIB_FILE
+# @REQUIRED
+# @DESCRIPTION:
+# Absolute path of this Libretro core's shared library.
+: ${LIBRETRO_CORE_LIB_FILE:="${S}/${LIBRETRO_CORE_NAME}_libretro.so"}
+
+case "${EAPI:-0}" in
+	6)
+		EXPORT_FUNCTIONS src_unpack src_prepare src_compile src_install
+		;;
+        *) die "EAPI=${EAPI} is not supported" ;;
+esac
+
+# @FUNCTION: libretro-core_src_unpack
+# @DESCRIPTION:
+# The libretro-core src_unpack function which is exported.
+#
+# This function retrieves the remote Libretro core info files.
+libretro-core_src_unpack() {
+	# If this is a live ebuild, retrieve this core's remote repository.
+	if [[ ${PV} == "9999" ]]; then
+		git-r3_src_unpack
+		# Add used commit SHA for version information, the above could also work.
+		LIBRETRO_COMMIT_SHA=$(git -C "${EGIT3_STORE_DIR}/${LIBRETRO_REPO_NAME//\//_}.git" rev-parse HEAD)
+	# Else, unpack this core's local tarball.
+	else
+		default_src_unpack
+	fi
+}
+
+# @FUNCTION: libretro-core_src_prepare
+# @DESCRIPTION:
+# The libretro-core src_prepare function which is exported.
+#
+# This function prepares the source by making custom modifications.
+libretro-core_src_prepare() {
+	local flags_modified=0
+	ebegin "Attempting to hack Makefiles to use custom-cflags"
+	for makefile in "${S}"/?akefile* "${S}"/target-libretro/?akefile*; do
+		# * Convert CRLF to LF
+		# * Expand *FLAGS to prevent potential self-references
+		# * Where LDFLAGS directly define the link version 
+		#   script append LDFLAGS and LIBS
+		# * Where SHARED is used to provide shared linking
+		#   flags ensure final link command includes LDFLAGS
+		#   and LIBS
+		# * Always use $(CFLAGS) when calling $(CC)
```

Comment:
can't we work with upstream to use a sane buildsystem instead of awful homebrew makefiles?

Reply:
Maybe? I'm not quite sure I'm ready to boil that ocean right now though... the problem exists in a dozen repositories, not all of which are under the control of the same organization.